src/coreclr/src/System.Private.CoreLib/src/System/Threading/Thread.CoreCLR.cs

```diff
@@ -486,19 +486,31 @@ private static int CalculateOptimalMaxSpinWaitsPerSpinIteration()
         [MethodImpl(MethodImplOptions.InternalCall)]
         private static extern int GetCurrentProcessorNumber();
 
-        // The upper bits of t_currentProcessorIdCache are the currentProcessorId. The lower bits of
-        // the t_currentProcessorIdCache are counting down to get it periodically refreshed.
-        // TODO: Consider flushing the currentProcessorIdCache on Wait operations or similar
-        // actions that are likely to result in changing the executing core
-        [ThreadStatic]
-        private static int t_currentProcessorIdCache;
+        // t_currentProcessorId lives in a separate class to make sure the class is fully initialized by the time we use the field
```

Comment:
There is no initializer for CoreIdCache. What class is this comment talking about?

Reply:
Accessing `t_currentProcessorId` gets noticeably more expensive in `SimpleCoreIdSpeedCheck` if it is in the same class. It looks like JIT takes slow path while class is still not done with static initialization.

I could be wrong about the actual reasons (would be interesting to know what really happened), but moving the field into a nested class made access times from static field initializer in the outer more reasonable.

---
In reply to: [352949830](https://github.com/dotnet/runtime/pull/467#discussion_r352949830) [](ancestors = 352949830)

Reply:
I will move entire caching/calibration logic to this class (as hinted in another comment). Then this comment will be no longer relevant.

---
In reply to: [352991722](https://github.com/dotnet/runtime/pull/467#discussion_r352991722) [](ancestors = 352991722,352949830)